apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/CsvReaderImpl.java

```diff
@@ -0,0 +1,497 @@
+    private int inputBufferStart = -1;
+    private int inputBufferEnd;
+
+    private boolean endOfLine;
+    private boolean endOfFile;
+
+    @Inject
+    public CsvReaderImpl(@Named("dataExportDir") Path dataExportPath) {
+        super.dataExportPath = Objects.requireNonNull(dataExportPath, "dataExportPath is NULL");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResultSet read(String inputFileName, String[] colNames, String charset) throws SQLException {
+        Objects.requireNonNull(inputFileName, "inputFileName is NULL");
+        assignNewFileName(inputFileName, true);
+        try {
+            return readResultSet(colNames);
+        } catch (IOException e) {
+            throw new SQLException("Exception reading (or not found) file '"
+                    + inputFileName + "' in path = " + super.dataExportPath, e);
+        }
+    }
+
+    private void assignNewFileName(String newFileName, boolean closeWhenAppend) {
+        Objects.requireNonNull(newFileName, "fileName is NULL");
+        this.fileName = newFileName;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResultSet read(Reader reader, String[] colNames) throws IOException {
+        Objects.requireNonNull(reader, "reader is NULL");
+        this.input = reader;
+        return readResultSet(colNames);
+    }
+
+    private ResultSet readResultSet(String[] colNames) throws IOException {
+        this.columnNames = colNames;
+        initRead();
+        SimpleResultSet result = new SimpleResultSet(this);
+        makeColumnNamesUnique();
+        int index = 0;
+        //reading from header like COLUMN_NAME(TYPE|PRECISION|SCALE)
+        for (String columnName : columnNames) { // reading header meta data
+            ColumnMetaData metaData = columnsMetaData[index];
+            if (metaData != null) {
+                // actual meta data
+                result.addColumn(columnName, metaData.getSqlTypeInt(), metaData.getPrecision(), metaData.getScale());
+            } else {
+                // default meta data values
+                result.addColumn(columnName, Types.VARCHAR, Integer.MAX_VALUE, 0);
+            }
+            index++;
+        }
+        return result;
+    }
+
+    private void initRead() throws IOException {
+        if (input == null) {
+            try {
+                InputStream in = DbUtils.newInputStream(
+                        this.dataExportPath,
+                        !this.fileName.contains(CSV_FILE_EXTENSION) ? this.fileName + CSV_FILE_EXTENSION : this.fileName
+                );
+                in = new BufferedInputStream(in, IO_BUFFER_SIZE);
+                input = new InputStreamReader(in, characterSet);
+            } catch (IOException e) {
+                close();
+                throw e;
+            }
+        }
+        if (!input.markSupported()) {
+            input = new BufferedReader(input);
+        }
+        input.mark(1);
```

Comment:
Do we require such compatibility?

Reply:
I would like do not changed that H2 borrowed code now.